docs/demo-weather-agent.md

```diff
@@ -0,0 +1,166 @@
+   Or use a custom repository accessible using the GitHub ID specified in your `.env` file.
+5. For **Git Branch or Tag**, use the default `main` branch (or select another as needed).
+6. Set **Protocol** to `acp`.
+7. Under [**Specify Source Subfolder**](http://kagenti-ui.localtest.me:8080/Import_New_Agent#specify-source-subfolder):
+   - Click `Select from examples`
+   - Choose: `acp/acp_weather_service`
+8. Click **Build New Agent** to deploy.
+
+---
+
+## Import New Tool
+
+To deploy the Weather Tool:
+
+1. Navigate to [Import New Tool](http://kagenti-ui.localtest.me:8080/Import_New_Tool#import-new-tool) in the UI.
+2. Select the same `<namespace>` as used for the agent.
+3. In the **Select Environment Variable Sets** section, select:
+   - `mcp-weather`
+   - `ollama`
+4. Use the same source repository:
+   <https://github.com/kagenti/agent-examples>
+5. Choose the `main` branch or your preferred branch.
+6. Set **Select Protocol** to `MCP`.
+7. Under **Specify Source Subfolder**:
+   - Select: `mcp/weather_tool`
+8. Click **Build New Tool** to deploy.
+
+---
+
+## Validate The Deployment
+
+Depending on your hosting environment, it may take some time for the deployment to complete.
+
+To verify that both the agent and tool are running:
+
+1. Open a terminal and connect to your Kubernetes cluster.
+2. Use the namespace you selected during deployment to check the status of the pods:
+
+   ```console
+   installer$ kubectl -n team1 get po
+   NAME                                  READY   STATUS    RESTARTS   AGE
+   acp-weather-service-8bb4644fc-4d65d   1/1     Running   0          1m
+   weather-tool-5bb675dd7c-ccmlp         1/1     Running   0          1m
+   ```
+
+3. Tail the logs to ensure both services have started successfully.
+   For the agent:
+
+   ```console
+    installer$ kubectl -n team1 logs -f acp-weather-service-8bb4644fc-4d65d
+    Defaulted container "acp-weather-service" out of: acp-weather-service, kagenti-client-registration (init)
+    INFO:     Started server process [18]
+    INFO:     Waiting for application startup.
+    INFO:     Application startup complete.
+    INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+    ```
+
+    For the tool:
+    ```console
+    installer$ kubectl -n team1 logs -f weather-tool-5bb675dd7c-ccmlp
+    Defaulted container "weather-tool" out of: weather-tool, kagenti-client-registration (init)
+    INFO:     Started server process [19]
+    INFO:     Waiting for application startup.
+    INFO:     Application startup complete.
+    INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+    ```
+
+4. Once you see the logs indicating that both services are up and running, you're ready to proceed to [Authorize the Agent and the Tool](#authorize-the-agent-and-the-tool).
+
+## Authorize the Agent and the Tool
+
+Once the agent and the tool are deployed, you need to authorize the agent to interact with the tool:
```

Comment:
What we do here right now is not technically "Authorizing" the agent (recall the tool does no validation right now)
Instead, I might adjust these texts to say we are "registering" the agent onto Keycloak.

Reply:
we don't need to worry too much because I suppose we will be updating this demo soon anyhow

Reply:
This is a foundation for further updates. We will revisit Authorization section again later.